### PR TITLE
Lvrubym 220: Add health report

### PR DESCRIPTION
**Pull request by skrypnyk88**
## JIRA

* [Main JIRA ticket](http://ssu-jira.softserveinc.com/browse/LVRUBYM-220)

## Code reviewers

- [x] @Mykhailo123 
- [x] @yaremakleputs 
- [x] @andriyarterchuk

### Second Level Review

- [ ] @4asof4ik

## Summary of issue

Need to add HealthReport

## Summary of change

- added health report

## Testing approach

Test with JSCS

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/js/common/resources/healthReport.resource.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    constants.name
+    ])
+  .factory('healthReportResource', healthReportResource);
+
+healthReportResource.$inject = ['$resource', 'globalSettings', 'currentGroupDay'];
+
+function healthReportResource($resource, globalSettings, currentGroupDay) {
+  return $resource(globalSettings.SERVER_URL_V1 + '/health_reports/:id.json',
+  {id: '@id',
+    group_id: currentGroupDay.group_id},
```
Comment (andriyarterchuk) [inline app/js/common/resources/healthReport.resource.js]: where date?

File: index.html [hunk 2]
```diff
@@ -9,6 +9,7 @@
   </head>
   <body>
     <main ui-view></main>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.0.0/jquery.min.js"></script>
```
Comment (andriyarterchuk) [inline index.html]: why you need this?

File: app/js/common/resources/healthReport.resource.js (hunk 1 above)
Comment (skrypnyk88) [inline app/js/common/resources/healthReport.resource.js]: Do we need it?

File: app/js/healthReport/components/healthReport.template.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    </button>
+  </div>
+  <div flex="30">
+    <textarea rows="5" cols="45" ng-model="report.report.health_note"
+                                 name="text"
+                                 class="text-area-health-note"></textarea>
+  </div>
+  <div flex="20" class="update" >
+    <button class="group-x update-note-button"
+            ng-click="$ctrl.healthReportUpdate(report); $ctrl.reload()">
```
Comment (4asof4ik) [inline app/js/healthReport/components/healthReport.template.html]: read about angular form and use ng-submit
$ctrl.healthReportUpdate(report.report)

File: app/js/healthReport/components/healthReport.component.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    HealthReport.getReports().then(
+      function(data) {
+        ctrl.reports = data;
+        ctrl.presence = ctrl.reports.length;
+      }
+    );
+  };
+
+  ctrl.healthReportUpdate = function(report) {
+    HealthReport.updateReports(report.report.health_note,
```
Comment (4asof4ik) [inline app/js/healthReport/components/healthReport.component.js]: report.health_note......................... read next comment

File: test/spec/healthReport.service.spec.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
+var healthVendor = require('./../../app/js/requirements.js');
+var HealthReport = require('./../../app/js/common/services/healthReport.service.js');
+var constantsModule = require('./../../app/js/common/constants.js');
+var valuesModule = require('./../../app/js/common/values.js');
+var toggleMessage =
+  require('./../../app/js/common/services/toggleMessage/toggleMessage.service.js');
```
Comment (4asof4ik) [inline test/spec/healthReport.service.spec.js]: in one line

File: test/spec/healthReport.service.spec.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+    $httpBackend.flush();
+
+    expect(JSON.stringify(response)).toEqual(JSON.stringify(reports));
+  }));
+
+  it('error message updated report', inject(function(toggleMessage,
+                                                     globalSettings,
+                                                     currentGroupDay) {
+    $httpBackend.whenPUT(globalSettings.SERVER_URL_V1 + '/health_reports/' +
```
Comment (4asof4ik) [inline test/spec/healthReport.service.spec.js]: empty line

File: test/spec/healthReport.service.spec.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+    $httpBackend.flush();
+
+    expect(JSON.stringify(response)).toEqual(JSON.stringify(errors));
+  }));
+
+  it('when updated report', inject(function(toggleMessage,
+                                            globalSettings,
+                                            currentGroupDay) {
+    $httpBackend.whenPUT(globalSettings.SERVER_URL_V1 + '/health_reports/' +
```
Comment (4asof4ik) [inline test/spec/healthReport.service.spec.js]: and here

File: app/js/common/services/auth.service.js [hunk 8]
```diff
@@ -123,12 +123,12 @@ function AuthFactory(
...
 
   function saveCurrentUser(response) {
-    currentUser.first_name = response.first_name;
-    currentUser.last_name = response.last_name;
-    currentUser.locale = response.locale;
-    currentUser.id = response.id;
-    currentUser.url = response.url;
-    currentUser.phone = response.phone;
-    currentUser.email = response.email;
+    for (var currentUserProp in currentUser) {
```
Comment (skrypnyk88) [inline app/js/common/services/auth.service.js]: no

**Pull request merged**